test/rubygems/test_gem_package.rb

```diff
@@ -140,6 +140,20 @@ def test_build_time_without_source_date_epoch
     ENV["SOURCE_DATE_EPOCH"] = epoch
   end
 
+  def test_build_time_ignores_invalid_source_date_epoch
+    ENV["SOURCE_DATE_EPOCH"] = "315532800"
```

Comment:
This environment variable should be restored after this test, like the other related tests.